### PR TITLE
fix : Realm access from incorrect thread.

### LINKED ISSUE #1758: App crash in the livefeed
**Actual Behaviour**

The app is crashing in the live feed after the latest series of merges.

**Expected Behaviour**

The live feed should work fine.

**Steps to reproduce the issue**

1. Open the app.
2. The progress bar seems to be be stuck at 85%.
3. Minimise the app.
4. Open the live feed tab from the Nav Drawer.

**LogCat for the issue**

screenshot from 2017-07-01 16-12-38

**Would you like to work on the issue?**

Yes

**Target Milestone to solve the issue**

[milestone 8](https://github.com/fossasia/open-event-android/milestones/31)

### PR DESCRIPTION
**Pull request by rohanagarwal94**
Fixes #1758 and #1757

Issue logs discovered after debugging only - 

screenshot from 2017-07-01 16-48-56

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
@iamareebjamal @arp95 @Shailesh351 @shreyasnbhat please review.

**Review by Shailesh351 — APPROVED** [review]
@rohanagarwal94 I think issue regarding app carsh in live feed is different.

**Comment by rohanagarwal94** [comment]
@shailesh351 even I thought the same. But to my surprise as and when the downloader issue got resolved, the live feed too didnt crash.

File: android/app/src/main/java/org/fossasia/openevent/dbutils/RealmDataRepository.java [hunk 1]
```diff
@@ -472,6 +472,7 @@ public Completable saveLocations(final List<Microlocation> locations) {
     //Session Types Section
 
     private void saveSessionTypesInRealm(List<SessionType> sessionTypes) {
+        Realm realm = Realm.getDefaultInstance();
         realm.executeTransaction(realm1 -> realm1.insertOrUpdate(sessionTypes));
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/dbutils/RealmDataRepository.java:476]: The solution is to not create a new instance. See other method bodies on how it is done

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/dbutils/RealmDataRepository.java (hunk 1 above)
Comment (rohanagarwal94) [reply android/app/src/main/java/org/fossasia/openevent/dbutils/RealmDataRepository.java:476]: @iamareebjamal I have seen other method bodies before coming to the solution. Please check it from your side.

Comment (iamareebjamal) [reply android/app/src/main/java/org/fossasia/openevent/dbutils/RealmDataRepository.java:476]: OK, thread safety in Realm has really taken every bit of it over.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**